### ISSUE TITLE
Bad block on morden

### ISSUE THREAD

**Issue by GregTheGreek** [etclabscore/go-ethereum#12]
Re: https://github.com/eth-classic/go-ethereum/issues/87

It seems that classic geth produced a bad block at 4729797

Unsure of block contents as noone we know produced transactions. Further investigation is required

**Comment by soc1c** [comment]
Transaction from Parity Ethereum

```
{
  blockHash: "0x438e4c5fc151d6ff45aad23281a67abdd70e1a5306383a349c1e12113a9179ca",
  blockNumber: 4729797,
  chainId: "0x3e",
  condition: null,
  creates: "0xb460c361dec258e1d1ca0dd4cebc9e4f014b57d0",
  from: "0xfb289c3d95fc768f42576e61e943bec290a9ce1f",
  gas: 1000000,
  gasPrice: 20000000000,
  hash: "0x9c86be2f3ed4a5102d3f8a9b766535500ff910bb421f701eefaa1eb4594eaa46",
  input: "0x6060604052341561000f57600080fd5b6040516020806103b58339810160405280805160008054600160a060020a03338116600160a060020a0319928316179092556001805492909316911617905550506103568061005f6000396000f3006060604052600436106100535763ffffffff60e060020a60003504166310c42eaf81146100555780634d63a5421461006c5780634fe870ab146100995780635bee6f92146100c85780638da5cb5b146100e7575b005b610053600160a060020a03600435166024356100fa565b610053600160a060020a03600435166024356fffffffffffffffffffffffffffffffff19604435166101c2565b34156100a457600080fd5b6100ac6102c2565b604051600160a060020a03909116815260200160405180910390f35b34156100d357600080fd5b610053600160a060020a03600435166102d1565b34156100f257600080fd5b6100ac61031b565b60005433600160a060020a039081169116148061017f5750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561016257600080fd5b5af1151561016f57600080fd5b5050506040518051151560011490505b151561018a57600080fd5b600160a060020a03821634820180156108fc0290604051600060405180830381858888f1935050505015156101be57600080fd5b5050565b60005433600160a060020a03908116911614806102475750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561022a57600080fd5b5af1151561023757600080fd5b5050506040518051151560011490505b151561025257600080fd5b82600160a060020a031663359d1f118334018360405160e060020a63ffffffff85160281526fffffffffffffffffffffffffffffffff1990911660048201526024016000604051808303818588803b15156102ac57600080fd5b5af115156102b957600080fd5b50505050505050565b600154600160a060020a031681565b60005433600160a060020a039081169116146102ec57600080fd5b6001805473ffffffffffffffffffffffffffffffffffffffff1916600160a060020a0392909216919091179055565b600054600160a060020a0316815600a165627a7a7230582046ee003a7d8c03a02f6ba3dfdb80862c30abaf5fcbb8d5cbfccde55270e6e2a3002900000000000000000000000072dab16d86da2e2c05d0bff39477f271b059dd0e",
  nonce: 1050277,
  publicKey: "0x29338f38f4b71d1f66834b97753bc6f7379c5dd27ab3d7c139380daaedbd0802f4416ef528dadcbb721dc3ea31be8f8ba72d51fb4c9c71fa20588d37c5aca4cc",
  r: "0x45ced644cf9c6645de1933c8ebfddd734c6605e050e1419e5d39d924424e31c1",
  raw: "0xf9042c831006a58504a817c800830f42408080b903d56060604052341561000f57600080fd5b6040516020806103b58339810160405280805160008054600160a060020a03338116600160a060020a0319928316179092556001805492909316911617905550506103568061005f6000396000f3006060604052600436106100535763ffffffff60e060020a60003504166310c42eaf81146100555780634d63a5421461006c5780634fe870ab146100995780635bee6f92146100c85780638da5cb5b146100e7575b005b610053600160a060020a03600435166024356100fa565b610053600160a060020a03600435166024356fffffffffffffffffffffffffffffffff19604435166101c2565b34156100a457600080fd5b6100ac6102c2565b604051600160a060020a03909116815260200160405180910390f35b34156100d357600080fd5b610053600160a060020a03600435166102d1565b34156100f257600080fd5b6100ac61031b565b60005433600160a060020a039081169116148061017f5750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561016257600080fd5b5af1151561016f57600080fd5b5050506040518051151560011490505b151561018a57600080fd5b600160a060020a03821634820180156108fc0290604051600060405180830381858888f1935050505015156101be57600080fd5b5050565b60005433600160a060020a03908116911614806102475750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561022a57600080fd5b5af1151561023757600080fd5b5050506040518051151560011490505b151561025257600080fd5b82600160a060020a031663359d1f118334018360405160e060020a63ffffffff85160281526fffffffffffffffffffffffffffffffff1990911660048201526024016000604051808303818588803b15156102ac57600080fd5b5af115156102b957600080fd5b50505050505050565b600154600160a060020a031681565b60005433600160a060020a039081169116146102ec57600080fd5b6001805473ffffffffffffffffffffffffffffffffffffffff1916600160a060020a0392909216919091179055565b600054600160a060020a0316815600a165627a7a7230582046ee003a7d8c03a02f6ba3dfdb80862c30abaf5fcbb8d5cbfccde55270e6e2a3002900000000000000000000000072dab16d86da2e2c05d0bff39477f271b059dd0e819fa045ced644cf9c6645de1933c8ebfddd734c6605e050e1419e5d39d924424e31c1a04af8aaf6e2a2b16598b3980bcf287fb3009dc52bdf793a01a225148c18a61150",
  s: "0x4af8aaf6e2a2b16598b3980bcf287fb3009dc52bdf793a01a225148c18a61150",
  standardV: "0x0",
  to: null,
  transactionIndex: 0,
  v: "0x9f",
  value: 0
}
```

Block by Parity-Ethereum

```
{
  author: "0x00d5842976c0d35d511dae0a36f9a2c9368ee8b0",
  difficulty: 530953590,
  extraData: "0x6675636b2074686520706f6c696365",
  gasLimit: 6901573,
  gasUsed: 327535,
  hash: "0x438e4c5fc151d6ff45aad23281a67abdd70e1a5306383a349c1e12113a9179ca",
  logsBloom: "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  miner: "0x00d5842976c0d35d511dae0a36f9a2c9368ee8b0",
  mixHash: "0xfad19eeebca388acb3d710f1551de56488efdbe4b243db10d8855e95d6c2af13",
  nonce: "0x3abcc64857c0f4a7",
  number: 4729797,
  parentHash: "0xc6c4a38b7238472999a62783e191399a9f2ff9089e258a3febc424287aa91f43",
  receiptsRoot: "0xd96459bc12f49d4023166247226b0e84a53ba7e7418dc4e955809ca934295f1f",
  sealFields: ["0xa0fad19eeebca388acb3d710f1551de56488efdbe4b243db10d8855e95d6c2af13", "0x883abcc64857c0f4a7"],
  sha3Uncles: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
  size: 1606,
  stateRoot: "0x155454f6e1fe47bf999b14c3616d56a203b21ffd00e0f2e0af0e3d935865f434",
  timestamp: 1564363903,
  totalDifficulty: 624501892982658,
  transactions: ["0x9c86be2f3ed4a5102d3f8a9b766535500ff910bb421f701eefaa1eb4594eaa46"],
  transactionsRoot: "0xc569c93b30be792f132072a16637e725fe2cf95b88486bedbfeb0b07c88fe226",
  uncles: []
}
```

**Comment by soc1c** [comment]
Transaction from Geth Classic

```
{
  blockHash: "0x118c7b9aa7dad9da4887b6043964b3fdb8dd5e00647952f229b50d6ca8ffe550",
  blockNumber: 4729797,
  chainId: 62,
  from: "0xfb289c3d95fc768f42576e61e943bec290a9ce1f",
  gas: 1000000,
  gasPrice: 20000000000,
  hash: "0x9c86be2f3ed4a5102d3f8a9b766535500ff910bb421f701eefaa1eb4594eaa46",
  input: "0x6060604052341561000f57600080fd5b6040516020806103b58339810160405280805160008054600160a060020a03338116600160a060020a0319928316179092556001805492909316911617905550506103568061005f6000396000f3006060604052600436106100535763ffffffff60e060020a60003504166310c42eaf81146100555780634d63a5421461006c5780634fe870ab146100995780635bee6f92146100c85780638da5cb5b146100e7575b005b610053600160a060020a03600435166024356100fa565b610053600160a060020a03600435166024356fffffffffffffffffffffffffffffffff19604435166101c2565b34156100a457600080fd5b6100ac6102c2565b604051600160a060020a03909116815260200160405180910390f35b34156100d357600080fd5b610053600160a060020a03600435166102d1565b34156100f257600080fd5b6100ac61031b565b60005433600160a060020a039081169116148061017f5750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561016257600080fd5b5af1151561016f57600080fd5b5050506040518051151560011490505b151561018a57600080fd5b600160a060020a03821634820180156108fc0290604051600060405180830381858888f1935050505015156101be57600080fd5b5050565b60005433600160a060020a03908116911614806102475750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561022a57600080fd5b5af1151561023757600080fd5b5050506040518051151560011490505b151561025257600080fd5b82600160a060020a031663359d1f118334018360405160e060020a63ffffffff85160281526fffffffffffffffffffffffffffffffff1990911660048201526024016000604051808303818588803b15156102ac57600080fd5b5af115156102b957600080fd5b50505050505050565b600154600160a060020a031681565b60005433600160a060020a039081169116146102ec57600080fd5b6001805473ffffffffffffffffffffffffffffffffffffffff1916600160a060020a0392909216919091179055565b600054600160a060020a0316815600a165627a7a7230582046ee003a7d8c03a02f6ba3dfdb80862c30abaf5fcbb8d5cbfccde55270e6e2a3002900000000000000000000000072dab16d86da2e2c05d0bff39477f271b059dd0e",
  nonce: 1050277,
  r: "0x45ced644cf9c6645de1933c8ebfddd734c6605e050e1419e5d39d924424e31c1",
  replayProtected: true,
  s: "0x4af8aaf6e2a2b16598b3980bcf287fb3009dc52bdf793a01a225148c18a61150",
  to: null,
  transactionIndex: 0,
  v: "0x9f",
  value: 0
}
```

Block from Geth Classic

```
{
  difficulty: 530953590,
  extraData: "0x",
  gasLimit: 6888140,
  gasUsed: 327535,
  hash: "0x118c7b9aa7dad9da4887b6043964b3fdb8dd5e00647952f229b50d6ca8ffe550",
  logsBloom: "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  miner: "0xc931b469a707e55d738fb46e16db65abcda037be",
  mixHash: "0xbf5457ae81d15b5e8d10370c5efdd24a99a90918298082f270d22d5f7443171b",
  nonce: "0x78ae1c0010aadbdc",
  number: 4729797,
  parentHash: "0xc6c4a38b7238472999a62783e191399a9f2ff9089e258a3febc424287aa91f43",
  receiptsRoot: "0xd96459bc12f49d4023166247226b0e84a53ba7e7418dc4e955809ca934295f1f",
  sha3Uncles: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
  size: 1591,
  stateRoot: "0x4bcd0fc60a213f583b2c4414f5a378ce5db555ad767d1a09a1e83359087b67ce",
  timestamp: 1564363903,
  totalDifficulty: 624501892982658,
  transactions: ["0x9c86be2f3ed4a5102d3f8a9b766535500ff910bb421f701eefaa1eb4594eaa46"],
  transactionsRoot: "0xc569c93b30be792f132072a16637e725fe2cf95b88486bedbfeb0b07c88fe226",
  uncles: []
}
```

**Comment by soc1c** [comment]
http://www.mergely.com/M08Eark9/

**Comment by GregTheGreek** [comment]
Here is the tx data:
```
0x6060604052341561000f57600080fd5b6040516020806103b58339810160405280805160008054600160a060020a03338116600160a060020a0319928316179092556001805492909316911617905550506103568061005f6000396000f3006060604052600436106100535763ffffffff60e060020a60003504166310c42eaf81146100555780634d63a5421461006c5780634fe870ab146100995780635bee6f92146100c85780638da5cb5b146100e7575b005b610053600160a060020a03600435166024356100fa565b610053600160a060020a03600435166024356fffffffffffffffffffffffffffffffff19604435166101c2565b34156100a457600080fd5b6100ac6102c2565b604051600160a060020a03909116815260200160405180910390f35b34156100d357600080fd5b610053600160a060020a03600435166102d1565b34156100f257600080fd5b6100ac61031b565b60005433600160a060020a039081169116148061017f5750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561016257600080fd5b5af1151561016f57600080fd5b5050506040518051151560011490505b151561018a57600080fd5b600160a060020a03821634820180156108fc0290604051600060405180830381858888f1935050505015156101be57600080fd5b5050565b60005433600160a060020a03908116911614806102475750600154600160a060020a031663fe9fbb803360405160e060020a63ffffffff8416028152600160a060020a039091166004820152602401602060405180830381600087803b151561022a57600080fd5b5af1151561023757600080fd5b5050506040518051151560011490505b151561025257600080fd5b82600160a060020a031663359d1f118334018360405160e060020a63ffffffff85160281526fffffffffffffffffffffffffffffffff1990911660048201526024016000604051808303818588803b15156102ac57600080fd5b5af115156102b957600080fd5b50505050505050565b600154600160a060020a031681565b60005433600160a060020a039081169116146102ec57600080fd5b6001805473ffffffffffffffffffffffffffffffffffffffff1916600160a060020a0392909216919091179055565b600054600160a060020a0316815600a165627a7a7230582046ee003a7d8c03a02f6ba3dfdb80862c30abaf5fcbb8d5cbfccde55270e6e2a3002900000000000000000000000072dab16d86da2e2c05d0bff39477f271b059dd0e
```

and here it is decoded:
https://pastebin.com/k3HuyPMf

**Comment by fish-sammy** [comment]
Not sure if this is related to the issue, but it seems that Parity and Geth have a chain split now on Morden. 

I am running `Geth v6.0.6` and `Parity v2.5.5` (both in a docker container with base image being `debian:stretch-slim`) on Morden testnet. The block hashes are the same for `4729276`, but starting from block `4729277`, here's what I got.

Geth
```
> web3.eth.getBlock(4729277).then(({ hash }) => console.log(hash))
> 0x8d02d003a416243acf173706fba49580974d909b1e8b3d6c58ec4a77c4972ce7
```

Parity
```
> web3.eth.getBlock(4729277).then(({ hash }) => console.log(hash))
> 0x67be5907a969faf8634ead6e296be52e3426cd4031be8436a4bdd147f0e9194
```

`4729277` is just several blocks after the scheduled fork block `4729274` on Morden. Is there a consensus issue going on between Geth and Parity now?

**Comment by GregTheGreek** [comment]
@sammy1991106 Thank you for the report!

This is fixed by #14 !

**Comment by GregTheGreek** [comment]
> `4729277` is just several blocks after the scheduled fork block `4729274` on Morden. Is there a consensus issue going on between Geth and Parity now?

The issue can actually be found [here](https://github.com/ethereum/EIPs/issues/2220) identifying that EIP-684 was missing from the required EIPS.

**Comment by austinabell** [comment]
closed with #21
